noxfile.py

```diff
@@ -10,28 +10,32 @@
 ).split(",")
 
 
-def deps(session):
+def deps(session, local_install):
     session.install("--upgrade", "setuptools", "pip")
-    session.install("-r", "requirements/dev.txt", ".", silent=True)
+    extra_flags = ["-e"] if local_install else []
+    session.install("-r", "requirements/dev.txt", *extra_flags, ".", silent=True)
```

Comment:
can you explain why we ever need to install with -e (aka local_install) here? I am not sure I understand the reasoning for adding this. can't docs and coverage use the installed parser?

Reply:
* `docs`: I don't think it matters, it works with both -- IIRC I set `local_install=True` just because the docs are only in the local install so it seemed natural
* `lint`: it makes more sense to me to use the local install because this is the code we are validating, that being said it also works with `local_install=False`
* `coverage`: this the one where it matters most, because `coverage run -m pytest` prepends `sys.path` with ".", so that it will end up using the local files anyway, which means that if you had a clean repo and run `nox -s coverage-3.8` it would fail due to missing the generated files locally (I'm going to rephrase the comment that explains it, the way it is written now suggests it is optional, and it's true in the sense that there could be other ways to fix this problem, but right now running it with `local_install=False` causes the problems I mentioned)

Reply:
Made the comment about the `coverage` session clearer in 723e7e1

Reply:
I would rather not make the noxfile more complicated if we can get away with it.
these kind of creeping complexity is not good.
Is this also a problem with Hydra?
generally I don't care about the coverage of the generated code at all and I think I am excluding it from the coverage reporting in Hydra.

Reply:
So, we have two threads of discussions on essentially the same topic (the other one is https://github.com/omry/omegaconf/pull/321#discussion_r513601312), let's continue the discussion in this one only:

* It's not a problem in Hydra but it's pure luck (see https://github.com/omry/omegaconf/pull/321#discussion_r464077526). I guess we could hack something to trigger a similar behavior but that seems, well, a hack ;)
* `coverage` requires `pip install -e .` because otherwise generated files will be missing from the local install, and `coverage` is always using the local install (even with `pip install .`). If we wanted to avoid that, the cleanest way would be to move the `omegaconf` folder into an `src` folder (that's another option I considered but I was trying to avoid it). And to be clear, the problem with the generated files missing isn't that we want coverage statistics for them, it's that code won't run at all without them.
* Although technically we could run everything on the local install (with `-e`), this is dangerous because it may hide future build issues

Recap of all options I can see right now:
1. Add a pytest entry point in setup.py to get the same behavior as Hydra, which would let `coverage` run on a regular (= non-local) install
2. Move all code into an `src` subfolder
3. Run all nox tests on the local install
4. Keep my fix